src/server/src/modules/job.ts

```diff
@@ -481,3 +480,99 @@ export const getApplicants = async (
 
     return getApplicantsResult;
 };
+
+/**
+ * @description Returns at most 20 newest jobs that match with student's department
+ *              and class standing.
+ * @param {number} studentId - id of student
+ * @param {number} page - page index
+ * @param {number} numOfItems - number of items per page
+ * @returns Promise
+ */
+export const getRecommendedJobs = async (
+    studentId: number
+) => {
+    const student = await Student.findOneOrFail({ where: { id: studentId } });
+
+    const jobApplications = await JobApplication.find({ where: { studentId } });
+    const appliedJobIds = jobApplications.map(
+        (jobApplication) => jobApplication.jobId
+    );
+
+    return getRepository(Job)
+        .createQueryBuilder('job')
+        .select([
+            'job',
+            'job.facultyMember',
+            'facultyMember.id',
+            'facultyMember.title',
+            'user.firstName',
+            'user.lastName',
+        ])
+        .leftJoin('job.facultyMember', 'facultyMember')
+        .leftJoin('facultyMember.user', 'user')
+        .leftJoinAndSelect('job.department', 'department')
+        .where('job.status = :jobStatus', {
+            jobStatus: 'Hiring',
+        })
+        .andWhere('job.id NOT IN (:...appliedJobIds)', {
+            // It causes a SQL parse error when an empty array is passed in.
+            appliedJobIds: appliedJobIds.length > 0 ? appliedJobIds : [-1],
+        })
+        .andWhere('(:departmentId < 0 OR job.departmentId = :departmentId)', {
+            departmentId: student.departmentId ?? -1
+        })
+        .andWhere(`(:classStanding = :null OR
+                    :classStanding = ANY(string_to_array(job.targetYears, :comma)))`, {
+            classStanding: student.classStanding ?? "NULL",
```

Comment:
what does "??" do? Never seen it before.

Reply:
student.classStanding ?? "NULL"

is same as

student.classStanding ? student.classStanding : "NULL"